.github/workflows/release.yml

```diff
@@ -0,0 +1,39 @@
+# References:
+# - https://github.com/cli/cli/blob/v2.32.1/.github/workflows/deployment.yml
+# - https://github.com/goreleaser/goreleaser-action/tree/v4.4.0#usage
+
+name: Release
+
+permissions:
+  contents: write
+
+on:
+  workflow_dispatch:
+    inputs:
+      tag_name:
+        required: true
+        type: string
+      go_version:
+        default: "1.20"
+        type: string
```

Comment:
Why do we provide a go-version? 🤔 
The `go.mod` file should be the source of truth what version we are using, no? 🤔 
(Talking about espeically Go 1.21, not of older versions (we are currently using 😁 ))

Reply:
Here we are specifying the go compiler version which can be different than go module version. Compilers >= code module version should be almost always successful due to go's compatibility promise. So using 'latest' as the value should also work but we are being more cautious here that is all. We do not want this to be one more cause among many other causes failure while doing release workflow for first time.

Reply:
Maybe my question was a bit missleading.

Why do we add the `go_version` into the `workflow_dispatch`?
Why not "hardcoded" into the `actions/setup-go@v4` action? 🤔 
What benefit do we have do specify a go version *when* doing the release? 🤔 
I guess this is also a bit confusing for people who do the release.
What should they put in?
1.20? 1.19? 1.21?
In case our `go.mod` points already to 1.21, why should we even add a version here.
The toolchain makes sure that 1.21 will be used anyways...